### ISSUE TITLE
Support nested layers in `layer-name-pattern` rule

### ISSUE THREAD

**Issue by jhae-de** [stylelint/stylelint-config-standard#372]
### What is the problem you're trying to solve?

Layers can be nested:

```css
@layer layer {
  @layer nested-layer {
  }
}
```

To append rules to the nested layer, the layers can be joined using dot notation.

```css
@layer layer.nested-layer {
  ...
}
```

Stylelint currently reports a problem here:

```
Expected layer name "layer.nested-layer" to be kebab-case (layer-name-pattern)
```

### What solution would you like to see?

Change the pattern of the `layer-name-pattern` rule to:

```
^([a-z][a-z0-9]*)([.-][a-z0-9]+)*$
```

**Comment by jeddy3** [comment]
@jhae-de Thanks for the report.

If I remember correctly, when we were implementing this rule in Stylelint we intended to support dot notation in this config, but it looks like we forgot to do it.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

**Comment by jhae-de** [comment]
@jeddy3 Thank you, I would take care of that and create a PR for it.
